Firestore/Example/App/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Copyright 2019 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+add_executable(macOS_Example_App MACOSX_BUNDLE
+  macOS/main.m
+  macOS/AppDelegate.m
+  macOS/AppDelegate.h
+)
+
+add_objc_flags(macOS_Example_App
```

Comment:
Note I have to call `add_objc_flags` with the same files as above, ideally we don't need to do this at all. Maybe we need a `objc_bundle` target?

Reply:
I'm not opposed, but we only have one of these at this point, so I'd suggest deferring this work until we have some idea how this works in more than one circumstance.

Note that this style of calling multiple functions to describe what's required to build a single target is common in CMake. Our concise helpers that make things look more like blaze/bazel are uncommon. 

In short, I think this is fine, but up to you.